kpi/models/import_export_task.py

```diff
@@ -689,15 +685,79 @@ def _run_task(self, messages):
             elif export_type == 'spss_labels':
                 export.to_spss_labels(output_file)
 
-        # Restore the FileField to its typical state
-        self.result.open('rb')
-        self.save(update_fields=['last_submission_time'])
+        self.result = absolute_filename
+
+        if not self.pk:
+            # In tests, exports are not saved into the DB before calling this
+            # method, thus we cannot update only specific fields.
+            self.save()
+        else:
+            self.save(update_fields=['result', 'last_submission_time'])
 
     def delete(self, *args, **kwargs):
         # removing exported file from storage
         self.result.delete(save=False)
         super().delete(*args, **kwargs)
 
+    def get_absolute_filename(self, filename: str) -> str:
+        """
+        Get absolute filename related to storage root.
+        """
+
+        storage_class = self.result.storage
+        filename = self.result.field.generate_filename(self, filename)
+
+        # We used to call `self.result.save(filename, ContentFile(b''))` before
```

Comment:
My take on this style of comment to consider and do with what you may. Concise is easier to read. You and I care deeply about the history and why changes are made. But is it really relevant what we used to do? I would reword to be more terse about needing this for Azure. If it's relevant to keep the old example, keep it but in the style of "This seems like a good idea but won't work". If not needed, that's even better and you can ditch it. Do keep the django bug reference if it's still relevant here.